src/gdclasses/OpenXRConfig.cpp

```diff
@@ -45,6 +51,55 @@ bool OpenXRConfig::keep_3d_linear() const {
 	}
 }
 
+int OpenXRConfig::get_view_config_type() const {
+	if (openxr_api == NULL) {
+		return 1;
+	} else {
+		XrViewConfigurationType config_type = openxr_api->get_view_configuration_type();
+		switch (config_type) {
+			case XR_VIEW_CONFIGURATION_TYPE_PRIMARY_MONO: {
+				return 0;
+			}; break;
+			case XR_VIEW_CONFIGURATION_TYPE_PRIMARY_STEREO: {
+				return 1;
+			}; break;
+			/* These are not (yet) supported, adding them in here for future reference
+			case XR_VIEW_CONFIGURATION_TYPE_PRIMARY_QUAD_VARJO: {
+				return 2;
+			}; break;
+			case XR_VIEW_CONFIGURATION_TYPE_SECONDARY_MONO_FIRST_PERSON_OBSERVER_MSFT: {
+				return 3;
+			}; break;
+			*/
+			default: {
+				// we don't support the others
+				Godot::print_error(String("Unsupported configuration type set: ") + String::num_int64(config_type), __FUNCTION__, __FILE__, __LINE__);
+				return 1;
+			}; break;
+		}
+	}
+}
```

Comment:
Shouldn't we just return the XR view configuration types values as is? What's the advantage for defining another set of values that map to them especially since the plan is to include as much of OpenXR as possible in Godot 4.x?

Reply:
I missed this comment which seems to make the case for it: `so our dropdown in the UI works properly`.

Seems fine by me if that's the only available option.

Reply:
Yeah it's just that for a enum dropdown in the UI you define the property like so:
```
	register_property<OpenXRConfig, int>("view_configuration", ... , GODOT_PROPERTY_HINT_ENUM, "Mono,Stereo");
```
So 0 = Mono, 1 = Stereo, etc.

Also at some point we will have to think of how we'll unify some of this in Godot 4. We'll likely end up with our own enumerations in the core of Godot for this that map to whatever systems and capabilities that are in place in the runtime. While I'd like to think everyone will switch to OpenXR and we don't have to worry about any other runtimes, I think that is wishful thinking.